lib/colors.sh

```diff
@@ -0,0 +1,31 @@
+# Utilities for working with colors
+#
+# Dependencies:
+#		- none
+
+#######################################
+# Accepts a single argument containing a color, returning a normalized color based
+# on the input
+#
+# If an integer between 1 and 3 characters long is provided, returns `colour[$1]`
+# otherwise returns the input unaltered
+#
+# Arguments:
+#   $1: The color to normalize
+#######################################
+__normalize_color() {
```

Comment:
👍 great idea to extract this, makes usage clearer in formatting, and with more features!